frontend/packages/dev-console/src/components/import/devfile/DevfileImportForm.tsx

```diff
@@ -23,7 +27,34 @@ const DevfileImportForm: React.FC<FormikProps<FormikValues> & DevfileImportFormP
 }) => {
   const { t } = useTranslation();
   const [, devfileParseError] = useDevfileServer(values, setFieldValue);
+  const searchParams = new URLSearchParams(window.location.search);
+  const gitRepoUrl = searchParams.get('gitRepo');
+  const devfileName = searchParams.get('devfileName');
+  const formType = searchParams.get('formType');
+  const [devfileSamples, setDevfileSamples] = React.useState<DevfileSample[]>();
+  const selectedSample = React.useMemo(
+    () => devfileSamples?.find((sample) => sample.name === devfileName),
+    [devfileSamples, devfileName],
+  );
   useDevfileDirectoryWatcher(values, setFieldValue);
+  useDevfileSource(gitRepoUrl, devfileName, formType, setFieldValue);
```

Comment:
You can actually use `useFormikContext` and that you won't have to send `setFieldValue`. Also he query params can be fetched inside the hook.

Reply:
Updated.